### PR TITLE
fix latest LLVM 11 compatibility issues

### PR DESCRIPTION
**Pull request by yhls**
- `CompositeType` class has been removed
- `FileLineInfoSpecifier::Default` has been renamed to `RawValue`

@vchuravy

### CODE REVIEW

**Comment by yhls** [comment]
- `SequentialType` class has been removed
- `CreateMemCpy` now takes `MaybeAlign`s instead of `unsigned`s

File: src/llvm-late-gc-lowering.cpp [hunk 1]
```diff
@@ -388,22 +388,29 @@ CountTrackedPointers::CountTrackedPointers(Type *T) {
...
+            count *= cast<VectorType>(T)->getNumElements();
     }
     if (count == 0)
         all = false;
 }
 
 unsigned getCompositeNumElements(Type *T) {
-    return isa<StructType>(T) ? T->getStructNumElements() : cast<SequentialType>(T)->getNumElements();
+    if (isa<StructType>(T))
+        return T->getStructNumElements();
```
Comment (vtjnash) [inline src/llvm-late-gc-lowering.cpp:409]:
```suggestion
        return T->getNumElements();
```

**Review by vtjnash — COMMENTED** [review]
```
/Users/julia/buildbot/worker/package_macos64/build/src/llvm-late-gc-lowering.cpp:414:1: warning: control may reach end of non-void function [-Wreturn-type]
}
^
1 warning generated.
In file included from /Users/julia/buildbot/worker/package_macos64/build/src/codegen.cpp:853:
./cgutils.cpp:1349:28: warning: variable 'num_elements' is used uninitialized whenever 'if' condition is false [-Wsometimes-uninitialized]
            else if (auto *VT = dyn_cast<VectorType>(T))
                           ^~
./cgutils.cpp:1351:17: note: uninitialized use occurs here
            if (num_elements == 0)
                ^~~~~~~~~~~~
./cgutils.cpp:1349:18: note: remove the 'if' if its condition is always true
            else if (auto *VT = dyn_cast<VectorType>(T))
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./cgutils.cpp:1346:34: note: initialize the variable 'num_elements' to silence this warning
            uint64_t num_elements;
                                 ^
                                  = 0
1 warning generated.
```
Need to fix the generated warnings, but otherwise lgtm

File: src/llvm-late-gc-lowering.cpp (hunk 1 above)
Comment (yhls) [reply src/llvm-late-gc-lowering.cpp:409]: To use `getNumElements`, I'd have to explicitly cast `T` to `StructType` like in the following cases--do you suggest that I do this?

**Comment by yhls** [comment]
I've fixed the warnings, and addressed the following change in LLVM

- deprecated overloads of `IRBuilder::CreateCall` and `CallInst::Create` have been removed

by making `prepare_call_in` in cgutils.cpp return a `FunctionCallee` instead of a `Value*`.

**Comment by yhls** [comment]
- `CallSite` class has been removed

**Review by vchuravy — COMMENTED** [review]
> To use getNumElements, I'd have to explicitly cast T to StructType like in the following cases--do you suggest that I do this?

Probably want to use a `dyn_cast` then.

File: src/cgutils.cpp [hunk 2]
```diff
@@ -1341,9 +1343,16 @@ static void emit_write_multibarrier(jl_codectx_t&, Value*, Value*);
 
 std::vector<unsigned> first_ptr(Type *T)
 {
-    if (isa<CompositeType>(T)) {
```
[the remark this answers is not in the document]
Comment (yhls) [reply src/cgutils.cpp:1344]: I don't think there's any way to construct a `CompositeType`, so it's effectively abstract, right?

**Comment by yhls** [comment]
> > To use getNumElements, I'd have to explicitly cast T to StructType like in the following cases--do you suggest that I do this?
> 
> Probably want to use a `dyn_cast` then.

I meant to say that using `getStructNumElements` allows us to avoid having to cast explicitly, as opposed to using `getNumElements` with a `dyn_cast`, so is probably preferable. But I'm pretty indifferent about one over the other here.

**Comment by vchuravy** [comment]
> I meant to say that using getStructNumElements allows us to avoid having to cast explicitly, as opposed to using getNumElements with a dyn_cast, so is probably preferable. But I'm pretty indifferent about one over the other here.

It looks like LLVM is removing all the convenience getters, and it is cheaper to do the `dyn_cast` then to do an `isa` check and then do the `cast` inside `getStructNumElements` (which will do another `isa` check.)

**Comment by yhls** [comment]
Ah, ok, changed to use `getNumElements`.

File: src/llvm-late-gc-lowering.cpp (hunk 1 above)
Comment (yhls) [reply src/llvm-late-gc-lowering.cpp]: Changed to use `getNumElements`.

**Pull request merged**